lte/gateway/c/session_manager/test/test_session_state_5g.cpp

```diff
@@ -0,0 +1,410 @@
+      create_static_rule_install("static-2"),
+  };
+  EXPECT_EQ(2, static_rule_installs.size());
+
+  PolicyRule dynamic_1 = create_policy_rule("dynamic-1", "", 0);
+  PolicyRule dynamic_2 = create_policy_rule_with_qos("dynamic-2", "", 0, 1);
+
+  // Create a DynamicRuleInstall rules
+  std::vector<DynamicRuleInstall> dynamic_rule_installs{
+      create_dynamic_rule_install(dynamic_1),
+      create_dynamic_rule_install(dynamic_2),
+  };
+  EXPECT_EQ(2, dynamic_rule_installs.size());
+
+  RulesToProcess pending_activation, pending_deactivation;
+  session_state->process_5g_rules_to_install(
+      static_rule_installs, dynamic_rule_installs, &pending_activation,
+      &pending_deactivation, &update_criteria);
+
+  EXPECT_EQ(4, pending_activation.size());
+
+  const std::string activate_rule1 = pending_activation[0].rule.id();
+  const std::string activate_rule2 = pending_activation[1].rule.id();
+
+  EXPECT_TRUE(activate_rule1 == "static-1" || activate_rule1 == "dynamic-1");
+  EXPECT_TRUE(activate_rule2 == "static-2" || activate_rule2 == "dynamic-2");
+
+  EXPECT_EQ(update_criteria.static_rules_to_install.size(), 2);
+  EXPECT_TRUE(update_criteria.static_rules_to_install.count("static-1"));
+  EXPECT_TRUE(update_criteria.static_rules_to_install.count("static-2"));
+
+  EXPECT_TRUE(session_state->is_static_rule_installed("static-1"));
+  EXPECT_TRUE(session_state->is_static_rule_installed("static-2"));
+
+  EXPECT_EQ(update_criteria.dynamic_rules_to_install.size(), 2);
+  EXPECT_EQ("dynamic-1", update_criteria.dynamic_rules_to_install[0].id());
+  EXPECT_EQ("dynamic-2", update_criteria.dynamic_rules_to_install[1].id());
+
+  session_state->remove_all_5g_rules_for_termination(&update_criteria);
+
+  EXPECT_EQ(update_criteria.static_rules_to_uninstall.size(), 1);
+  EXPECT_EQ(update_criteria.dynamic_rules_to_uninstall.size(), 2);
+
+  std::vector<std::string> rules_out{};
+  std::vector<std::string>& rules_out_ptr = rules_out;
+
+  session_state->get_dynamic_rules().get_rule_ids(rules_out_ptr);
+  EXPECT_EQ(rules_out_ptr.size(), 0);
+
+  EXPECT_FALSE(session_state->is_dynamic_rule_installed("dynamic-1"));
+  EXPECT_FALSE(session_state->is_dynamic_rule_installed("dynamic-2"));
+
+  std::string mkey;
+  // searching for non-existent rule should fail
+  EXPECT_FALSE(
+      session_state->get_dynamic_rules().get_monitoring_key_for_rule_id(
+          "dynamic-1", &mkey));
+  EXPECT_FALSE(
+      session_state->get_dynamic_rules().get_monitoring_key_for_rule_id(
+          "dynamic-2", &mkey));
+  // deleting an already deleted rule should fail
+  EXPECT_FALSE(
+      session_state->get_dynamic_rules().remove_rule("dynamic-1", &dynamic_1));
+  EXPECT_FALSE(
+      session_state->get_dynamic_rules().remove_rule("dynamic-2", &dynamic_2));
+}
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  FLAGS_logtostderr = 1;
+  FLAGS_v           = 10;
+  return RUN_ALL_TESTS();
+}
+
+}  // namespace magma
```

Comment:
`test_marshal_unmarshal` is pretty important because it can catch issues on format that would be overlooked otherwise. I see none of the 5g  fields are covered by that test.

Please, add a test_marshal_unmarshal for 5g case too taking into consideration fields like `fsm_state` `current_version`, `pdr_list`

Reply:
Yes. This part is missing. I will add for it.

Reply:
lets add `test_marshal_unmarshal`

Reply:
Yes, The test case is added for marshal_unmarshal